src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java

```diff
@@ -140,20 +140,21 @@ private static boolean canReturnOnlyBooleanLiteral(DetailAST ast) {
     }
 
     /**
-     * Returns if an AST is a return statement with a boolean literal.
+     * Returns if an AST is a return or a yield statement with a boolean literal.
      *
      * <p>Returns {@code true} iff ast represents
      * <pre>
-     * return true/false;
+     * return/yield true/false;
      * </pre>
      *
      * @param ast the syntax tree to check
-     * @return if ast is a return statement with a boolean literal.
+     * @return if ast is a return or a yield statement with a boolean literal.
      */
     private static boolean isBooleanLiteralReturnStatement(DetailAST ast) {
```

Comment:
I find it odd you updated the javadocs to make it clear it is a return or yield, but then you leave the method as `ReturnStatement`.

Reply:
done